### ISSUE TITLE
GKD generalized JSD implementation question

### ISSUE THREAD

**Issue by lidh15** [huggingface/trl#2202]
Hi, in trl GKD generalized JSD implementation we saw log prob mixture of teacher and student, while in the original paper it seems to mix the prob before applying log https://github.com/huggingface/trl/blob/main/trl/trainer/gkd_trainer.py#L186-L188
is this discrepancy a known improvement to the original paper? it really changes the behavior: say beta=0.5 by default, and student has a tiny prob close to 0 on a logit while teacher has a prob nearly 1, when mixing prob, the interpolated value will be about 0.5, but when mixing log prob, log prob of the tiny value will dominate and the interpolated value will be a large negative.

**Comment by kashif** [comment]
good point @lidh15 i think i had the logits interpolated and then while converting everything to log-probs i must have moved it around... I think this is a bug at first glance, let me double check and report back

**Comment by kashif** [comment]
good catch yes indeed mixtures and logs are not commutative!
